src/structure/group.rs

```diff
@@ -10,6 +10,7 @@ use super::{entity::Entity, guid::Guid};
 pub struct Group<T> {
     guid: Guid,
     marker: std::marker::PhantomData<T>,
+    endpoints: Vec<Endpoint>,
```

Comment:
I wasn't sure about this part, which I was I had left it out for now. Figure 8.5 certainly indicates that a `Group` should contain multiple `Endpoint`s, but I wasn't sure if this meant they should be stored directly in the group, or if the relationship was captured conceptually by each `Endpoint` having the same `Guid` prefix as the `Group` that contains them.
I was going to wait until I began implementing the behaviour/discovery modules to see which fell out naturally.

That said, I'm very open to being convinced!

Reply:
> it's a very different beast.

Right, I see that. The `InnerParticipant` is definitely getting low-level. https://github.com/jhelovuo/RustDDS/blob/4c10adf0bc7ce4efd5096903e03954ba145332fb/src/dds/pubsub.rs#L369

However, this shouldn't be the equivalent type here. There is a DDS DomainParticipant and an RTPS Participant. They are one-to-one AFAIK but the DDS DomainParticipant is the user facing type. See page 21 of the spec:

> RTPS Participant is the container of RTPS Group entities which contain Endpoint entities. The RTPS 
Participant maps to a DDS DomainParticipant.

Reply:
`Group` , as used in the RTPS spec, is an abstraction meaning "either a Publisher or a Subscriber".

AFAIK it has practical consequences only if you are implementing the QoS features "Group Ordered Access" or "Coherent Sets". As long as you do not support those, then grouping (e.g.) Readers under a Subscriber has little or no meaning. Group membership of Readers/Writers is not communicated over RTPS unless those QoS features are used.

All the Readers could be directly contained within a DomainParticipant with no intermediate Subscriber layer - and analogously for writing side. That is why RustDDS code has no concept of a Group.